test/test_sanitizer_log_parser.py

```diff
@@ -94,3 +108,30 @@ def make_key(line: Dict[str, str]) -> Tuple[Tuple[str, str], ...]:
 
     for line in sanitizer_log_parser_fixture.report_csv:
         assert line == expected_line_by_key.pop(make_key(line))
+
+
+def test_xml_has_testcase_output(sanitizer_log_parser_fixture) -> None:
+    if sanitizer_log_parser_fixture.resource_name == 'no_errors':
```

Comment:
This is fragile, it'd be better to either test invariant (number of errors should always be >= 0) or have keep pairs of files (one log file, one csv or xml file) to compare them.

This test cannot be edited easily by another developer because you're injecting in the test case knowledge about the log file (if test file == somefile.log, then it will have two warnings, four errors and 1 race condition).

This is better avoided because if the test fail, it will be impossible for us to know what's wrong (are those assumptions really correct? what if I want to update the log file?)

Reply:
Changed this test and the next one with the same problem to be fixture type agnostic.